components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -45,14 +114,15 @@ class Experiments(
      * saves them to local storage
      */
     @Synchronized
-    fun updateExperiments(): Boolean {
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun updateExperiments(): Boolean {
         if (!experimentsLoaded) {
             loadExperiments()
         }
         return try {
-            val serverExperiments = source.getExperiments(experimentsResult)
+            val serverExperiments = source!!.getExperiments(experimentsResult)
```

Comment:
Note that `!!` here will crash if `source = null`, and exception won't be trapped by the `catch` below. You might want to consider doing this:

```
source?.let {
  val serverExperiments = it.getExperiments(experimentsResult)
  experimentResult = serverExperiments
  storage?.save(serverExperiments)
} ?: logger.error("No source available")
```

This way the only throw due to `getExperiments` or, potentially, `save`.